src/core/ComplexFilter/index.tsx

```diff
@@ -117,10 +118,6 @@ export default function ComplexFilter<
   );
 
   function handleClick(event: React.MouseEvent<HTMLElement>) {
-    if (multiple) {
-      setPendingValue(value as DefaultMenuSelectOption[]);
```

Comment:
This is the root cause of resetting `pendingValue` prematurely, given the usage of `<ClickAwayListener />` now